migrations/20200811174954-create-accounts-for-workspaces.js

```diff
@@ -0,0 +1,177 @@
+const https = require('https');
+const axios = require('axios');
+
+require('dotenv').config({
+  path: '../.env'
+});
+
+/**
+ * Mutation for creating account
+ */
+const MUTATION_CREATE_ACCOUNT = `
+  mutation AccountCreateMutation($input: AccountInput!) {
+    account {
+      create(input: $input) {
+        recordId
+      }
+    }
+  }
+`;
+
+/**
+ * @file Create accounts for workspaces
+ */
+module.exports = {
+  async up(db, client) {
+    /**
+     * Use one transaction for all requests
+     */
+    const session = client.startSession();
+
+    const accountingApiConfig = {
+      baseURL: process.env.CODEX_ACCOUNTING_URL,
+      tlsVerify: process.env.TLS_VERIFY,
+      tlsCaCertPath: process.env.TLS_CA_CERT,
+      tlsCertPath: process.env.TLS_CERT,
+      tlsKeyPath: process.env.TLS_KEY
+    }
+
+    let httpsAgent = null;
+
+    if (accountingApiConfig.tlsVerify) {
+      httpsAgent = new https.Agent({
+        ca: fs.readFileSync(accountingApiConfig.tlsCaCertPath || ''),
+        cert: fs.readFileSync(accountingApiConfig.tlsCertPath || ''),
+        key: fs.readFileSync(accountingApiConfig.tlsKeyPath || ''),
+      });
+    }
+
+    const apiInstance = axios.create({
+      baseURL: accountingApiConfig.baseURL,
+      timeout: 1000,
+      httpsAgent: httpsAgent,
+    });
+
+    console.log('Start to creating accounts for workspaces...');
+
+    try {
+      await session.withTransaction(async () => {
+        /**
+         * Get all workspaces
+         */
+        const workspaces = await db.collection('workspaces').find().toArray();
+
+        /**
+         * Each workspace should have a plan
+         */
+        await Promise.all(workspaces.map(async workspace => {
+          const workspaceId = workspace._id;
+
+          /**
+           * Skip workspaces, which already have plan
+           */
+          if (workspace.accountId) {
+            return Promise.resolve();
+          }
+          console.log(`Workspace ${workspaceId} has no accountId`);
+
+          console.log(accountingApiConfig.baseURL);
+
+          const response = await apiInstance.post(accountingApiConfig.baseURL, {
+            query: MUTATION_CREATE_ACCOUNT,
+            variables: {
+              input: {
+                name: workspace.name,
```

Comment:
это имя счета? Тут точно только имя воркспейса или какое-то еще слово должно быть?

Reply:
> Account name (for example, "Cashbook")

просто бирочка

Reply:
Может быть добавим какую-нибудь пометку, что аккаунт относится к воркспейсам, чтобы можно было, если это понадобится, найти все такие аккаунты через regex. Например:
```suggestion
                name: `${workspace.name}-workspace`,
```